diff --git a/src/recyclerview/RecyclerView.tsx b/src/recyclerview/RecyclerView.tsx
index 3c25e7e8..0cda6c68 100644
--- a/src/recyclerview/RecyclerView.tsx
+++ b/src/recyclerview/RecyclerView.tsx
@@ -374,7 +374,10 @@ const RecyclerViewComponent = <T,>(
    */
   const validateItemSize = useCallback(
     (index: number, size: RVDimension) => {
-      const layout = recyclerViewManager.getLayout(index);
+      const layout = recyclerViewManager.tryGetLayout(index);
+      if (layout === undefined) {
+        return;
+      }
       const width = Math.max(
         Math.min(layout.width, layout.maxWidth ?? Infinity),
         layout.minWidth ?? 0
diff --git a/dist/recyclerview/RecyclerView.js b/dist/recyclerview/RecyclerView.js
index 6799c8a6..0d9d7f47 100644
--- a/dist/recyclerview/RecyclerView.js
+++ b/dist/recyclerview/RecyclerView.js
@@ -241,7 +241,10 @@ const RecyclerViewComponent = (props, ref) => {
      */
     const validateItemSize = useCallback((index, size) => {
         var _a, _b, _c, _d;
-        const layout = recyclerViewManager.getLayout(index);
+        const layout = recyclerViewManager.tryGetLayout(index);
+        if (layout === undefined) {
+            return;
+        }
         const width = Math.max(Math.min(layout.width, (_a = layout.maxWidth) !== null && _a !== void 0 ? _a : Infinity), (_b = layout.minWidth) !== null && _b !== void 0 ? _b : 0);
         const height = Math.max(Math.min(layout.height, (_c = layout.maxHeight) !== null && _c !== void 0 ? _c : Infinity), (_d = layout.minHeight) !== null && _d !== void 0 ? _d : 0);
         if (areDimensionsNotEqual(width, size.width) ||
